x/storage/keeper/payment.go

```diff
@@ -309,12 +346,35 @@ func (k Keeper) ChargeViaBucketChange(ctx sdk.Context, bucketInfo *storagetypes.
 		return fmt.Errorf("get new bucket bill failed: %s %w", bucketInfo.BucketName, err)
 	}
 
-	// charge according to bill change
-	err = k.ApplyBillChanges(ctx, prevBill, newBill)
-	if err != nil {
-		ctx.Logger().Error("charge via bucket change failed", "bucket", bucketInfo.BucketName, "err", err.Error())
-		return err
+	if ctx.IsUpgraded(upgradetypes.Pawnee) {
```

Comment:
Can we refactor the code a bit? 
1. move line 336 to the inner of `if ctx.IsUpgraded(upgradetypes.Pawnee)`
2. in the `if ctx.IsUpgraded(upgradetypes.Pawnee)` we just set the value of `isPreviousBucketSuspended`. And out of the `if` we can take different actions based on `isPreviousBucketSuspended`.